File: springapp/src/main/java/com/examly/springapp/serviceimpl/AdmissionService.java
package com.examly.springapp.serviceimpl; 
import java.util.List; 
import java.util.Optional; 
import com.examly.springapp.model.Admission; 
import com.examly.springapp.model.AdmissionStatus; 
import com.examly.springapp.model.Student; 
import com.examly.springapp.model.User; 
import com.examly.springapp.repository.AdmissionRepository; 
import com.examly.springapp.repository.AdmissionStatusRepository; 
import com.examly.springapp.repository.StudentRepository; 
import com.examly.springapp.repository.UserRepository; 
import org.springframework.stereotype.Service; 
import java.util.Set; 
import org.springframework.beans.factory.annotation.Autowired; 

@Service 
public class AdmissionService { 

    @Autowired 
	private AdmissionRepository admrepo; 
	
	@Autowired 
	private AdmissionStatusRepository asr; 
	
	@Autowired 
	private UserRepository userrepo; 
	 
	@Autowired 
	private StudentRepository stur; 
     
    public User saveAdmission(Long userid,Admission admission){ 
        User user=userrepo.findById(userid).get(); 
        AdmissionStatus adms=new AdmissionStatus(1, null); 
	    admission.setAdmissionStatus(adms); 
		user.setAdmission(admission); 
		 
		return this.userrepo.save(user);
    } 

    public Admission editAdmission(Admission admission){ 
         
        return this.admrepo.save(admission); 
    } 
 
    public List<Admission> getAdmissions(){ 
        return this.admrepo.findAll(); 
    } 

    public Admission getAdmission(int admissionId){ 
	    
        return this.admrepo.findById(admissionId).get(); 
    } 

    public Set<Admission> getApproved(){ 
        return this.admrepo.getApplication("APPROVED"); 
    } 
    public Set<Admission> getPending(){ 
        return this.admrepo.getApplication("PENDING"); 
    } 
    public Set<Admission> getRejected(){ 
        return this.admrepo.getApplication("REJECTED"); 
    } 

    public Admission approve(int admissionId){ 
        Admission admission=admrepo.findById(admissionId).get(); 
		AdmissionStatus adms=new AdmissionStatus(2, null); 
		admission.setAdmissionStatus(adms);	 
		return this.admrepo.save(admission); 
    } 

    public Admission reject(int admissionId){ 
        Admission admission=admrepo.findById(admissionId).get(); 
		AdmissionStatus adms=new AdmissionStatus(3, null); 
		admission.setAdmissionStatus(adms);	 
		return this.admrepo.save(admission); 
    } 


} 
 

File: angularapp/src/app/services/admission.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class AdmissionService {
  constructor(private http:HttpClient) { }

  public url="http://35.170.243.30:8080"

  public saveAdmission(userid,admission){
    return this.http.post(`${this.url}/saveAdmission/${userid}`,admission);
  }
   
  public getallPendingAdmissions(){
    return this.http.get(`${this.url}/pending`);
  }
  public getallRejectedAdmissions(){
    return this.http.get(`${this.url}/rejected`);
  }
  public getallApprovedAdmissions(){
    return this.http.get(`${this.url}/approved`);
  }

  public getAdmission(admissionId){
    return this.http.get(`${this.url}/getadmission/${admissionId}`);
  }

  public approveApplication(admissionId){
    return this.http.put(`${this.url}/approve/${admissionId}`,'');
  }

  public rejectApplication(admissionId){
    return this.http.put(`${this.url}/reject/${admissionId}`,'');
  
  }
  public getCourses(instituteId){
    return this.http.get(`${this.url}/getcourses/instituteid/${instituteId}`);
  }

  public editAdmission(admission){
    return this.http.put(`${this.url}/editadmission`,admission);
  }
}
